wg-data-protection/charter.md

```diff
@@ -0,0 +1,78 @@
+# WG Data Protection Charter
+
+This charter adheres to the [wg-governance] guidance as well as
+the general conventions described in the [Kubernetes Charter README] and
+the Roles and Organization Management outlined in [sig-governance], where
+applicable to a Working Group.
+
+
+## Scope
+
+Data protection in Kubernetes context typically involves backup and recovery
+of two types of entities:
+* Kubernetes object resources
+* Persistent volume data
+We consider it a complicated and layered problem, including backup and recovery
+at persistent volume level, application level, and cluster level. Part of the
+working group’s charter is to define what Kubernetes native constructs are
+required to achieve these goals.
+
+The Data Protection Working Group is organized with the goal of providing
+a cross SIG forum to discuss how to support data protection in Kubernetes,
+identify missing functionality, and work together to design features that
+are needed to achieve the goal.
+
+### In scope
+
+* High level discussions on what it means to support data protection in Kubernetes at different levels and how to do it.
+* Design discussions on specific topics related to data protection and disaster recovery support.
+* Document results of discussions and investigations in a linkable medium.
+
+Potential design topics include, but are not limited to the following:
+* Read data from a snapshot without creating a new volume.
+* Volume backups
+* Data populator
+* Retrieve diffs between two snapshots (block and file level)
+* Consistency volume groups (group snapshot)
+* Application snapshot, backup, and recovery
+* Data protection policy
+* Data protection workflows
+
+### Out of Scope
+
+* Design discussions not related to data protection is out of scope. For example,
+  how to enhance volume placement and how to report volume health belong to SIG
+  Storage and would not be a focus area of this WG. Workload API designs for
+  StatefulSet and Deployment belong to SIG Apps, however, this WG would be interested
+  in figuring out how to backup and recover a StatefulSet or Deployment.
+* This is a working group so it does not own code. Design discussions for
+  a specific feature including KEP reviews can happen in the working group
+  but KEP approvals and code implementation will be owned by SIG-Apps or
+  SIG-Storage.
+
+
+## Stakeholders
+
+Stakeholders for this working group include members in the following SIGs:
+* SIG Apps
+* SIG Storage
+* SIG Auth (consult SIG Auth from security aspect)
+
+Stakeholders also include backup vendors who want to provide data protection
+support in Kubernetes and end users who want to use data protection applications.
+
+
+## Disband criteria
+
+It is our intent to disband once data protection support is mature in
```

Comment:
From the wg-governance document:

>How does the group know when the problem solving process is completed, and it is time for the Working Group to dissolve?

I think I would ask here, what does "mature" mean in this context?

Reply:
I think "mature" means people in the WG believe there is no more missing functionality for Data Protection in Kubernetes and the Data Protection workflow is well defined and well supported.

Reply:
It would be good if we could crisp up the statement.  It seems like the WG wants to establish a cross-SIG roadmap for a set of enhancements required to meet the identified requirements.  Is it possible to say that the WG will define that document, and will disband once the roadmap is agreed upon with each of the SIG stakeholders?  At that point, its a matter of execution against the roadmap which should happen with the code owning bodies (i.e. SIGS)?  I am trying to think about how to use a crisper definition of mature oriented around a work product that is not code.

Reply:
Statement is modified to address Derek's comments.  Thanks.